bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareDeviceHandler.java

```diff
@@ -0,0 +1,255 @@
+
+    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");
+
+    private static final float BATTERY_FULL_VOLTAGE = 4 * 1.5f; // 4x AA batteries of 1.5V each
+    private static final float BATTERY_EMPTY_VOLTAGE = 4.2f; // 4x AA batteries of 1.05V each
+    private static final float LOW_BATTERY_THRESHOLD = 4 * 1.1f;
+
+    private final Logger logger = LoggerFactory.getLogger(SurePetcareDeviceHandler.class);
+
+    public SurePetcareDeviceHandler(Thing thing, SurePetcareAPIHelper petcareAPI) {
+        super(thing, petcareAPI);
+        logger.debug("Created device handler for type {}", thing.getThingTypeUID().getAsString());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateThingCache.getValue();
+        } else if (channelUID.getId().startsWith(DEVICE_CHANNEL_CURFEW_BASE)) {
+            handleCurfewCommand(channelUID, command);
+        } else {
+            switch (channelUID.getId()) {
+                case DEVICE_CHANNEL_LOCKING_MODE:
+                    if (command instanceof StringType) {
+                        synchronized (petcareAPI) {
+                            SurePetcareDevice device = petcareAPI.getDevice(thing.getUID().getId());
+                            if (device != null) {
+                                String newLockingModeIdStr = ((StringType) command).toString();
+                                try {
+                                    Integer newLockingModeId = Integer.valueOf(newLockingModeIdStr);
+                                    petcareAPI.setDeviceLockingMode(device, newLockingModeId);
+                                    updateState(DEVICE_CHANNEL_LOCKING_MODE,
+                                            new StringType(device.status.locking.modeId.toString()));
+                                } catch (NumberFormatException e) {
+                                    logger.warn("Invalid locking mode: {}, ignoring command", newLockingModeIdStr);
+                                } catch (SurePetcareApiException e) {
+                                    logger.warn(
+                                            "Error from SurePetcare API. Can't update locking mode {} for device {}",
+                                            newLockingModeIdStr, device);
+                                }
+                            }
+                        }
+                    }
+                    break;
+                case DEVICE_CHANNEL_LED_MODE:
+                    if (command instanceof StringType) {
+                        synchronized (petcareAPI) {
+                            SurePetcareDevice device = petcareAPI.getDevice(thing.getUID().getId());
+                            if (device != null) {
+                                String newLedModeIdStr = ((StringType) command).toString();
```

Comment:
You don't need to cast an object to call `toString()`.
```suggestion
                                String newLedModeIdStr = command.toString();
```

Reply:
done